api/v1alpha1/ndbserver_types.go

```diff
@@ -20,26 +20,35 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // NDBServerSpec defines the desired state of NDBServer
 type NDBServerSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// Foo is an example field of NDBServer. Edit ndbserver_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// +kubebuilder:validation:Required
+	Server string `json:"server"`
+	// +kubebuilder:validation:Required
+	CredentialSecret string `json:"credentialSecret"`
+	// +kubebuilder:default:=false
+	// +optional
+	// Skip server's certificate and hostname verification
+	SkipCertificateVerification bool `json:"skipCertificateVerification"`
 }
 
 // NDBServerStatus defines the observed state of NDBServer
 type NDBServerStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Status      string                           `json:"status"`
+	LastUpdated string                           `json:"lastUpdated"`
+	Databases   map[string]NDBServerDatabaseInfo `json:"databases"`
+	Counter     Counter                          `json:"counter"`
+}
+
+type Counter struct {
+	Database int `json:"database"`
```

Comment:
Easier to just keep "Counter" as an integer field inside NDBServerStatus?

Reply:
For now just putting the counter as an int field may be easier, yes!
But I purposely avoided it due to the following reason: Soon we'll need to add TM (and Snapshot) counters too, following the int field pattern we'll have 3 counters. The way reconcile works right now is to check if status and it's altered copy are deep equal (with the exception of the "Counter" key). We ignore the entire counter object from status deep equal check this way and would not have to do it for individual fields. 
Let me know how you feel about this.

Reply:
Could we rename "database_reconcile_counter" ? I got confused if this is the number of databases in NDB

Reply:
I don't want to repeat the word counter since it is already being used in the uber struct. I can however rename the counter to ReconcileCounter to address the confusion. WDYT?